examples/decoding/plot_ssd_spatial_filters.py

```diff
@@ -0,0 +1,442 @@
+            obtained from continous data or 3D array obtained from epoched
+            data.
+        y : None | array, shape (n_samples,)
+                    Used for scikit-learn compatibility.
+        Returns
+        -------
+        self : instance of SSD
+            Returns the modified instance.
+        """
+        self._check_X(X)
+        if X.ndim == 2:
+            X_aux = X[self.picks_, :].copy()
+
+        elif X.ndim == 3:
+            X_aux = X[:, self.picks_, :].copy()
+        else:
+            raise NotImplementedError()
+
+        X_signal = filter_data(
+            X_aux, self.info['sfreq'], **self.filt_params_signal)
+        X_noise = filter_data(
+            X_aux, self.info['sfreq'], **self.filt_params_noise)
+
+        X_noise -= X_signal 
+        if X.ndim == 3:
+            X_signal = np.hstack(X_signal)
+            X_noise = np.hstack(X_noise)
+
+        cov_signal = _regularized_covariance(
+            X_signal, reg=self.estimator, method_params=self.cov_method_params,
+            rank=self.rank, info=self.info)
+        cov_noise = _regularized_covariance(
+            X_noise, reg=self.estimator, method_params=self.cov_method_params,
+            rank=self.rank, info=self.info)
+   
+        eigvals_, eigvects_ = eigh(cov_signal, cov_noise)
+        # sort in descencing order
+        ix = np.argsort(eigvals_)[::-1]
+        self.eigvals_ = eigvals_[ix]
+        self.filters_ = eigvects_[:, ix]
+        self.patterns_ = np.linalg.pinv(self.filters_)
+        return self
+
+    def transform(self, X, y=None):
+        """Estimate epochs sources given the SSD filters.
+
+        Parameters
+        ----------
+        X : array, shape (n_channels, n_times) | shape (n_epochs,
+                n_channels, n_times)
+            The input data from which to estimate the SSD. Either 2D array
+            obtained from continous data or 3D array obtained from epoched
+            data.
+        y : None | array, shape (n_samples,)
+                    Used for scikit-learn compatibility.
+
+        Returns
+        -------
+        X_ssd : instance of Raw, Epochs or np.array
+            The processed data.
+        """
+        self._check_X(X)
+        if self.filters_ is None:
+            raise RuntimeError('No filters available. Please first call fit')
+        if X.ndim == 2:
+            X_ssd = np.dot(self.filters_.T, X[self.picks_])
+        elif X.ndim == 3:
+            # project data on source space
+            X_ssd = np.empty_like(X)
+            for ii, x in enumerate(X[:, self.picks_]):
+                X_ssd[ii] = np.dot(self.filters_.T, x)
+        else:
+            raise NotImplementedError()
+
+        if self.sort_by_spectral_ratio:
+            self.spec_ratio, self.sorter_spec = self.spectral_ratio_ssd(
+                ssd_sources=X_ssd)
+            self.filters_ = self.filters_[:, self.sorter_spec]
+            self.patterns_ = self.patterns_[self.sorter_spec]
```

Comment:
@vpeterson these two lines are inducing some vertigo to me.
Do you agree that we could just delete them?
We just sort the output while not updating inplace the fitted filters/patterns.

Reply:
yeah, that is a cleaner solution !